### PR TITLE
[TAS-158] ✨ Revamp index page

### PR DESCRIPTION
**Pull request by nwingt**

### CODE REVIEW

File: src/pages/nft/class/_classId/_nftId.vue [hunk 1]
```diff
@@ -393,7 +393,10 @@ export default {
     const description =
       this.nftDescription || this.$t('nft_details_page_description');
     const ogImage =
-      this.nftImageURL || 'https://liker.land/images/og/default.jpg';
+      this.NFTImageUrl ||
+      `${EXTERNAL_HOST}/images/og/${
+        this.$i18n.locale === 'en' ? 'default-en.png' : 'default.png'
```
[the remark this answers is not in the document]
Comment (nwingt) [reply src/pages/nft/class/_classId/_nftId.vue]: Updated

File: src/layouts/default.vue [hunk 2]
```diff
@@ -113,7 +103,26 @@ export default {
...
+        {
+          hid: 'og:description',
+          property: 'og:description',
+          content: this.$t('og_description'),
+        },
+        {
+          hid: 'og:image',
+          property: 'og:image',
+          content: `${EXTERNAL_HOST}/images/og/${
+            this.$i18n.locale === 'en' ? 'default-en.png' : 'default.png'
```
[the remark this answers is not in the document]
Comment (nwingt) [reply src/layouts/default.vue]: Updated

File: src/pages/index.vue [hunk 3]
```diff
@@ -1,420 +1,2202 @@
...
-      )
-    );
-    this.latestClassIds = latestClasses
-      .slice(0, NFT_CLASS_DISPLAY_COUNT)
-      .map(c => c.id);
-    this.trendingClassIds = this.filterNFTClassesByOwner(trendingClasses);
+    },
+  },
+  mounted() {
+    window.addEventListener('resize', this.handleResize);
```
[the remark this answers is not in the document]
Comment (nwingt) [reply src/pages/index.vue]: Updated

File: src/pages/nft/class/_classId/index.vue [hunk 4]
```diff
@@ -362,7 +362,10 @@ export default {
     const description =
       this.NFTDescription || this.$t('nft_details_page_description');
     const ogImage =
-      this.NFTImageUrl || 'https://liker.land/images/og/default.jpg';
+      this.NFTImageUrl ||
+      `${EXTERNAL_HOST}/images/og/${
+        this.$i18n.locale === 'en' ? 'default-en.png' : 'default.png'
```
[the remark this answers is not in the document]
Comment (nwingt) [reply src/pages/nft/class/_classId/index.vue]: Updated

File: src/pages/index.vue [hunk 5]
```diff
@@ -1457,10 +1457,15 @@ export default {
     },
   },
   mounted() {
-    window.addEventListener('resize', this.handleResize);
+    this.resizeListener = window.addEventListener('resize', this.handleResize);
```
Comment (williamchong) [inline src/pages/index.vue:1460]: Should define `resizeListener` in `data`?

File: src/pages/index.vue [hunk 6]
```diff
@@ -1457,10 +1457,15 @@ export default {
...
     this.handleResize();
     this.animate();
   },
+  beforeDestroy() {
+    if (this.resizeListener) {
+      window.removeEventListener(this.resizeListener);
```
Comment (williamchong) [inline src/pages/index.vue:1466]: Best to set `this.resizeListener` as null after removing it

**Review by williamchong — CHANGES_REQUESTED** [review]

File: src/pages/index.vue (hunk 5 above)
Comment (nwingt) [reply src/pages/index.vue:1460]: I think no need to put it in data for reactivity

Comment (williamchong) [reply src/pages/index.vue:1460]: Best to define it somewhere clearly as a class atteibute though

Comment (nwingt) [reply src/pages/index.vue]: Defined in `$options`

**Review by williamchong — APPROVED** [review]

**Pull request merged**